chainer/links/model/vision/vgg.py

```diff
@@ -0,0 +1,305 @@
+            conv5_2=Convolution2D(512, 512, 3, 1, 1, **kwargs),
+            conv5_3=Convolution2D(512, 512, 3, 1, 1, **kwargs),
+            fc6=Linear(512 * 7 * 7, 4096, **kwargs),
+            fc7=Linear(4096, 4096, **kwargs),
+            fc8=Linear(4096, 1000, **kwargs),
+        )
+        if pretrained_model == 'auto':
+            _retrieve(
+                'VGG_ILSVRC_16_layers.npz',
+                'http://www.robots.ox.ac.uk/%7Evgg/software/very_deep/'
+                'caffe/VGG_ILSVRC_16_layers.caffemodel',
+                self)
+        elif pretrained_model is not None:
+            npz.load_npz(pretrained_model, self)
+
+        self.functions = OrderedDict([
+            ('conv1_1', [self.conv1_1, relu]),
+            ('conv1_2', [self.conv1_2, relu]),
+            ('pool1', [_max_pooling_2d]),
+            ('conv2_1', [self.conv2_1, relu]),
+            ('conv2_2', [self.conv2_2, relu]),
+            ('pool2', [_max_pooling_2d]),
+            ('conv3_1', [self.conv3_1, relu]),
+            ('conv3_2', [self.conv3_2, relu]),
+            ('conv3_3', [self.conv3_3, relu]),
+            ('pool3', [_max_pooling_2d]),
+            ('conv4_1', [self.conv4_1, relu]),
+            ('conv4_2', [self.conv4_2, relu]),
+            ('conv4_3', [self.conv4_3, relu]),
+            ('pool4', [_max_pooling_2d]),
+            ('conv5_1', [self.conv5_1, relu]),
+            ('conv5_2', [self.conv5_2, relu]),
+            ('conv5_3', [self.conv5_3, relu]),
+            ('pool5', [_max_pooling_2d]),
+            ('fc6', [self.fc6, relu, dropout]),
+            ('fc7', [self.fc7, relu, dropout]),
+            ('fc8', [self.fc8, relu]),
+            ('prob', [softmax]),
+        ])
+
+    @property
+    def available_layers(self):
+        return list(self.functions.keys())
+
+    @classmethod
+    def convert_caffemodel_to_npz(cls, path_caffemodel, path_npz):
+        """Converts a pre-trained caffemodel to a chainer model.
+
+        Args:
+            path_caffemodel (str): Path of the pre-trained caffemodel.
+            path_npz (str): Path of the converted chainer model.
+        """
+
+        # As CaffeFunction uses shortcut symbols,
+        # we import CaffeFunction here.
+        from chainer.links.caffe.caffe_function import CaffeFunction
+        caffemodel = CaffeFunction(path_caffemodel)
+        npz.save_npz(path_npz, caffemodel, compression=False)
+
+    def __call__(self, x, layers=['prob'], test=True):
+        """Computes all the feature maps specified by ``layers``.
+
+        Args:
+            x (~chainer.Variable): Input variable.
+            layers (list of str): The list of layernames you want to extract.
+            test (bool): If ``True``, it runs in test mode.
+
+        Returns:
+            Dictionary of ~chainer.Variable: The directory in which
+            the key contains the layer name and the value contains
+            the corresponding variable.
```

Comment:
ditto
Maybe for clarity: 
corresponding variable -> corresponding feature map variable